contracts/zero-ex/contracts/src/features/ITransformERC20.sol

```diff
@@ -0,0 +1,127 @@
+/*
+
+  Copyright 2020 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-erc20/contracts/src/v06/IERC20TokenV06.sol";
+import "../transformers/IERC20Transformer.sol";
+import "../external/IPuppet.sol";
+
+
+/// @dev Feature to composably transform between ERC20 tokens.
+interface ITransformERC20 {
+
+    /// @dev Defines a transformation to run in `transformERC20()`.
+    struct Transformation {
+        // The transformation handler.
+        // Can receive the entire balance of `tokens`.
+        IERC20Transformer transformer;
+        // RLP-encoded deployment nonce of the deployer when `transformer`
+        // was deployed.
+        // This is used to verify that the transformer was deployed
+        // by a contract we trust.
+        bytes rlpNonce;
```

Comment:
The nonce isn't a known value. It increments for every contract. If a client screws it up, the tx just reverts.

Reply:
Yeah I was thinking along the lines of hardcoding the unique nonce into each contract before it's deployed. Then having an automatic check before or after the call, either by querying the transformer or standardizing the return data. Ex:
```
require(
    isZeroExTransformer(transformer.getNonce()),
    "ONLY_ZERO_EX_TRANSFORMERS"
);
transformer.transform(...);
```
OR
```
returnData = transformer.transform(...);
require(
    isZeroExTransformer(returnData.getNonce()),
    "ONLY_ZERO_EX_TRANSFORMERS"
);
```

Reply:
I like the latter. One less call. Just returning the nonce alone should be sufficient to indicate success since the RLP encoding is never empty and always unique.